src/ixdat/plotters/ms_plotter.py

```diff
@@ -397,6 +482,438 @@ def _parse_overloaded_inputs(
+                    from T_low_start to T_low_end missing representation of the high
+                    values in the middle of the axis.
+
+                    - If 'linear' sorting is specified all spectras obtained
+                        are sorted linearly from lowest v_name_value to highest v_name.
+                        When data is assymetric from scanning up or down in v_name this
+                        leads to abrupt looking figures since two non similair spectras
+                        are obtained at similair v_name value and hence plotted next to
+                        eachother.
+
+            kwargs: extra key-word args are passed on to matplotlib's plot()
+        """
+
+        if logplot is None:
+            logplot = not mol_lists and not mass_lists
+
+        if not axes:
+            if ms_data == "top":
+                n_bottom = 1
+                n_top = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 0
+                ms_spec_axes = 1
+            else:
+                n_top = 1
+                n_bottom = 2 if (mass_lists or mol_lists) else 1
+                ms_axes = 1
+                ms_spec_axes = 0
+
+            axes = self.new_two_panel_axes(
+                n_bottom=n_bottom,
+                n_top=n_top,
+                emphasis=emphasis,
+            )
+
+        measurement = measurement or self.measurement
+
+        if (
+            mass_list
+            or mass_lists
+            or mol_list
+            or mol_lists
+            or hasattr(measurement, "mass_list")
+        ):
+            # then we have MS data!
+            self.ms_plotter.plot_vs(
+                x_name=vs_name,
+                measurement=measurement,
+                axes=[axes[ms_axes], axes[2]]
+                if (mass_lists or mol_lists)
+                else [axes[ms_axes]],
+                tspan=tspan,
+                tspan_bg=tspan_bg,
+                remove_background=remove_background,
+                mass_list=mass_list,
+                mass_lists=mass_lists,
+                mol_list=mol_list,
+                mol_lists=mol_lists,
+                unit=unit,
+                x_unit=vs_unit,
+                logplot=logplot,
+                logdata=logdata,
+                legend=legend,
+                **kwargs,
+            )
+
+        # To plot heat plot.
+        # First get all values for v_name at all spectrum times
+        field = measurement.spectrum_series.field
+        _data = field.data.copy()
+
+        _t = field.axes_series[0].t
+        _v = measurement.grab_for_t(item=vs_name, t=_t)
+
+        if tspan:
+            # create t_mask from tspan
+            t_mask = np.logical_and(tspan[0] < _t, _t < tspan[-1])
+            # apply t_mask to field.data and vs_name.data
+            _data = _data[t_mask]
+            _v = _v[t_mask]
+
+        if isinstance(sort_spectra, str):
```

Comment:
This code got more complicated, which was not the intention. I do not like that sort_spectra can be so many different types. I like even less that there is no docstring explaining how sort_spectra is interpreted.

Reply:
I might misunderstand you, but line 748 the sort_spectra is described including uninteded behaviour in specific usecases?

Reply:
true. I missed that. Still quite complex but if it works and it's documented, I'm happy :)